src/amo/components/HomepageShelves/index.js

```diff
@@ -28,6 +31,12 @@ type InternalProps = {|
 
 export const HOMESHELVES_ENDPOINT_COLLECTIONS = 'collections';
 export const HOMESHELVES_ENDPOINT_SEARCH = 'search';
+export const HOMESHELVES_ENDPOINT_RANDOM_TAG = 'random-tag';
+
+function getTagFromUrl(api_url: string): string {
+  const { query } = url.parse(api_url, true);
+  return query && typeof query === 'object' ? query.tag : 'undefined';
```

Comment:
I think this could be rewritten as `return query && query[tag]` which would accomplish the same thing. I'm not even sure if you need the first `query`. Can `url.parse` return an object without a `query` property?

Reply:
apparently (from the failing [CI logs](https://app.circleci.com/pipelines/github/mozilla/addons-frontend/6287/workflows/f4d63311-dea2-4e3e-b52a-2de4b84e4671/jobs/44059?invite=true#step-105-58) query can be `null`, a `string`, or an `object`.  The url package doesn't seem to have type annotation so I guess Flow has worked that out itself.

Reply:
Fair enough. Thanks for checking.